pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -725,7 +724,7 @@ func (nc *Controller) monitorNodeStatus() error {
 					}
 					continue
 				}
-				exists, err := nc.nodeExistsInCloudProvider(types.NodeName(node.Name))
+				exists, err := nc.nodeExistsInCloudProvider(node)
```

Comment:
We should work with sig-node to remove this case from this controller. It is the only piece of cloud specific code in this controller. If we remove this call then the controller becomes CP agnostic.